jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -259,6 +264,16 @@ private Span setTagAsObject(String key, Object value) {
     return this;
   }
 
+  /**
+   * uses {@link #setTagAsObject} to only insert tags that don't have special significance
+   */
+  private void sanitizeAndSetTags(Map<String, Object> tags) {
```

Comment:
> one can never set a span.kind:server on the Span object. It is always discarded after setting the isRPC field to true.

This method is called from the constructor, so "after" is not applicable. And we do not discard any tags; they may not be stored as tags explicitly, but they are certainly "stored" one way or another. Nothing is being "sanitized" here.

I would just inline this code in the constructor, since it's never reused.